Mergin/switch_sources_dialog.py

```diff
@@ -0,0 +1,202 @@
+import os
+import yaml
+from dataclasses import dataclass, field
+import typing
+import re
+
+import psycopg2
+
+from qgis.core import (
+    QgsProject,
+    QgsDataSourceUri,
+    QgsMapLayer,
+    Qgis,
+    QgsVectorLayer,
+)
+from qgis.gui import QgsFileWidget, QgisInterface
+from qgis.PyQt import uic
+from qgis.PyQt.QtWidgets import QWizard, QLineEdit
+
+
+base_dir = os.path.dirname(__file__)
+ui_select_dbsync_page, base_select_dbsync_page = uic.loadUiType(
+    os.path.join(base_dir, "ui", "ui_switch_datasources_select_dbsync.ui")
+)
+ui_select_qgsproject_page, base_select_qgsproject_page = uic.loadUiType(
+    os.path.join(base_dir, "ui", "ui_switch_datasources_select_updated_project.ui")
+)
+
+
+DBSYNC_PAGE = 0
+QGS_PROJECT_PAGE = 1
+
+
+@dataclass
+class Connection:
+    driver: str
+    db_connection_info: str
+    db_schema: str
+    sync_file: str
+
+    valid: bool = False
+    db_tables: typing.List[str] = field(init=False)
+
+    def __post_init__(self):
+        try:
+            conn = psycopg2.connect(self.db_connection_info)
+        except Exception as e:
+            return
+        cur = conn.cursor()
+        cur.execute(f"SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname = '{self.db_schema}'")
+        self.db_tables = [x[0] for x in cur.fetchall()]
```

Comment:
if db_tables is only used in convert_to_postgresql_layer(), do we need to have it as a member variable?